### PR TITLE
Consolidating serialization tests.

### PR DESCRIPTION
**Pull request by SGuyGe**
@shmao @khdang @zhenlan

### CODE REVIEW

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs [hunk 1]
```diff
@@ -1326,19 +1715,36 @@ private static T SerializeAndDeserialize<T>(T value, string baseline, DataContra
...
-            dcjs.WriteObject(ms, value);
-            ms.Position = 0;
+            try
+            {
+                dcjs.WriteObject(ms, value);
+                ms.Position = 0;
+            }
+            catch
+            {
+                throw;
```
Comment (JonHanna) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: What's the advantage of this catch-throw?

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs [hunk 2]
```diff
@@ -1326,19 +1715,36 @@ private static T SerializeAndDeserialize<T>(T value, string baseline, DataContra
...
             string actualOutput = new StreamReader(ms).ReadToEnd();
             ms.Position = 0;
             Utils.CompareResult result = Utils.Compare(baseline, actualOutput, false);
 
-            Assert.True(result.Equal, string.Format("{1}{0}Test failed for input: {2}{0}Expected: {3}{0}Actual: {4}",
-                Environment.NewLine, result.ErrorMessage, value, baseline, actualOutput));
+            if (!result.Equal && !skipStringCompare)
+            {
+                throw new Exception(string.Format("Test failed for input : {0}\r\nError:{1}", value, result.ErrorMessage));
+            }
```
Comment (JonHanna) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: Why not `if(!skipStringCompar) Assert.True(…)`?

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by stephentoub** [comment]
The tests are failing:
```
DataContractSerializerTests.DCS_DuplicatedKeyDateTimeOffset [FAIL]
        System.Exception : Test failed for input : 1/2/2013 3:11:05 AM +00:00
EXEC : warning : XML comparision is also failing [d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Runtime.Serialization.Xml\tests\System.Runtime.Serialization.Xml.Tests.csproj]
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Runtime.Serialization.Xml\tests\DataContractSerializer.cs(1881,0): at DataContractSerializerTests.SerializeAndDeserialize[T](T value, String baseline, DataContractSerializerSettings settings, Func`1 serializerFactory, Boolean skipStringCompare)
           d:\j\workspace\dotnet_corefx_windows_debug_prtest\src\System.Runtime.Serialization.Xml\tests\DataContractSerializer.cs(1692,0): at DataContractSerializerTests.DCS_DuplicatedKeyDateTimeOffset()
```

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs (hunk 1 above)
Comment (SGuyGe) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: There used to be a Console.Writeline in catch block for debugging purpose, which was removed. So try/catch here really has no advantage, will remove it.

File: src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs (hunk 2 above)
Comment (SGuyGe) [inline src/System.Runtime.Serialization.Json/tests/DataContractJsonSerializer.cs]: Good suggestion, will use that.

**Comment by SGuyGe** [comment]
Updating code and will create a new PR after rebasing.

**Pull request closed without merging**